ios/Configuration/PartialPaymentParser.swift

```diff
@@ -0,0 +1,36 @@
+//
+// Copyright (c) 2024 Adyen N.V.
+//
+// This file is open source and available under the MIT license. See the LICENSE file for more info.
+//
+
+@_spi(AdyenInternal) import Adyen
+
+public struct PartialPaymentParser {
+
+    private var dict: [String: Any]
+
+    init(configuration: [String: Any]) {
+        self.init(configuration: configuration as NSDictionary)
+    }
+
+    public init(configuration: NSDictionary) {
```

Comment:
Any specific reason to use `NSDictionary`?

Reply:
ReactNative [operates with NSDictionary](https://reactnative.dev/docs/legacy/native-modules-ios#argument-types) since it is bridging via Obj-c

Reply:
Got it!

I see there are 25 occurrences of casting to `NSDictionary`, and I see there is an internal  `init(configuration: [String: Any])`. Would it be possible to remove excessive casting from unit tests part?

Reply:
Fixed [aced6be](https://github.com/Adyen/adyen-react-native/pull/546/commits/aced6bed518921f5a72f20fa6448494acdd54a8a)